src/controllers/cartsController.ts

```diff
@@ -0,0 +1,193 @@
+import { Request, Response } from 'express';
+import Database from '../database/index';
+import { sendResponse } from '../utils';
+import { cartProductInterface } from '../database/models/cart';
+import { calculateTotal, formatCartItems } from '../utils/cartsCalculations';
+
+interface UserInterface {
+  id: string;
+  name: string;
+  email: string;
+  role: string;
+}
+
+export const addItemToCart = async (req: Request, res: Response) => {
+  const { id } = req.user as UserInterface;
+
+  try {
+    const product = req.body as cartProductInterface;
+    const validProduct = await Database.Product.findOne({
+      where: { id: product.productId },
+    });
+    if (!validProduct) {
+      return sendResponse<null>(res, 404, null, 'Product not found');
+    }
+    if (validProduct.quantity < product.quantity) {
+      return sendResponse<null>(res, 404, null, 'Quantity Exceeds Stock');
+    }
+    const cartExist = await Database.Cart.findOne({ where: { userId: id } });
+    if (!cartExist) {
+      const cart = Database.Cart.build({
+        userId: id,
+        products: [product],
+        totalprice: validProduct.price * product.quantity,
+      });
+      await cart.save();
+      return sendResponse(res, 200, cart, 'Product added to cart');
+    }
+    const findProduct = cartExist.products.find(
+      p => p.productId === product.productId
+    );
+    if (findProduct) {
+      return sendResponse<null>(
+        res,
+        404,
```

Comment:
if product is exist it's doesn't mean it's 404 no we must have special status code